src/components/Export.tsx

```diff
@@ -0,0 +1,358 @@
+import { CheckCircleOutlined } from "@ant-design/icons";
+
+type ExportButtonProps = {
+  totalFrames: number;
+  setFrame: (frame: number) => void;
+  currentFrame: number;
+  startRecording: (options: Partial<RecordingOptions>) => void;
+  stopRecording: () => void;
+  defaultImagePrefix?: string;
+  disabled?: boolean;
+};
+
+export const TEST_ID_EXPORT_ACTION_BUTTON = "export-action";
+export const TEST_ID_OPEN_EXPORT_MODAL_BUTTON = "open-export-modal";
+
+const defaultProps: Partial<ExportButtonProps> = {
+  defaultImagePrefix: "image",
+  disabled: false,
+};
+
+const HorizontalDiv = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 6px;
+  flex-wrap: wrap;
+`;
+
+const VerticalDiv = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
+`;
+
+const CustomRangeDiv = styled(HorizontalDiv)`
+  & input {
+    width: 70px;
+    text-align: right;
+  }
+`;
+
+function clamp(value: number, min: number, max: number): number {
+  return Math.min(max, Math.max(value, min));
+}
+
+/**
+ * A single Export button that opens up an export modal when clicked. Manages starting and stopping
+ * an image sequence recording, resetting state when complete.
+ */
+export default function Export(inputProps: ExportButtonProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<ExportButtonProps>;
+
+  const theme = useContext(AppThemeContext);
+
+  const enum ExportMode {
+    ALL,
+    CURRENT,
+    CUSTOM,
+  }
+
+  // Static convenience method for creating simple modals + notifications.
+  // Used here for the cancel modal and the success notification.
+  // Note: notification API seems to only place notifications at the top-level under the
+  // <body> tag, which causes some issues with styling.
+  const { modal, notification } = App.useApp();
+  const modalContextRef = useRef<HTMLDivElement>(null);
+
+  const [isLoadModalOpen, setIsLoadModalOpen] = useState(false);
+  const [isRecording, setIsRecording] = useState(false);
+  const [isPlayingCloseAnimation, setIsPlayingCloseAnimation] = useState(false);
+
+  const [exportMode, setExportMode] = useState(ExportMode.ALL);
+  const [customMin, setCustomMin] = useState(0);
+  const [customMax, setCustomMax] = useState(props.totalFrames - 1);
+  const [imagePrefix, setImagePrefix] = useState(props.defaultImagePrefix);
+  const [useDefaultImagePrefix, setUseDefaultImagePrefix] = useState(true);
+  const [frameIncrement, setFrameIncrement] = useState(1);
+
+  const [originalFrame, setOriginalFrame] = useState(props.currentFrame);
+  const [percentComplete, setPercentComplete] = useState(0);
+
+  // If dataset changes, update the max range field with the total frames.
+  useEffect(() => {
+    setCustomMax(props.totalFrames - 1);
+  }, [props.totalFrames]);
+
+  useEffect(() => {
+    if (useDefaultImagePrefix) {
+      setImagePrefix(props.defaultImagePrefix);
+    }
+  }, [props.defaultImagePrefix, useDefaultImagePrefix]);
```

Comment:
I'd recommend using a deriver rather than an effect here:

```typescript
const getImagePrefix = () => useDefaultImagePrefix ? props.defaultImagePrefix : imagePrefix;
```

I believe it's recommended to avoid using effects to update state, since state updates will trigger a second render and a second dependency check for all your effects.

Reply:
Is there a difference between making this a function vs. a constant value?